packages/x-components/src/components/animations/create-directional-animation-factory.ts

```diff
@@ -19,23 +19,23 @@ export type AnimationOrigin =
  *
  * @internal
  */
-export function createDirectionalAnimationFactory(animationName: string) {
+export function createDirectionalAnimationFactory(
+  animationName: string
+): ReturnType<typeof defineComponent> {
   return (animationOrigin: AnimationOrigin = 'top') =>
     defineComponent({
       name: `transition-${animationName}-${animationOrigin}`,
       inheritAttrs: false,
-      setup(_, { attrs, listeners, slots }) {
+      setup(_, { attrs, slots }) {
         return () =>
           h(
-            'transition',
+            Transition,
             {
-              props: {
-                name: `x-${animationName}--${animationOrigin} x-${animationName}-`,
-                ...attrs
-              },
-              on: listeners
+              name: `x-${animationName}--${animationOrigin} x-${animationName}-`,
+              ...attrs
```

Comment:
Removing `inheritAttrs: false,` and `attrs` binding, it would be the same, wouldn't it? 🤔 
`Transition` would be the root node and the data would be linked to it directly